### PR TITLE
[Documentation] Adding link to Postgres upgrade article

### LINKED ISSUE #11248: PostgreSQL: Unclear upgrade instructions regarding identity generation (AUTO vs IDENTITY vs SEQUENCE) leads to unusable database
### BC Break Report

|    Q        |   A
|------------ | ------
| BC Break    | yes
| Version     | 3.0.0

#### Summary

I have this mapping:
```php
#[ORM\Id, ORM\Column(), ORM\GeneratedValue()]
private ?int $id = null;
```

After upgrading to DBAL 4 and ORM 3, running `php bin/console make:migration` generates:

```php
$this->addSql('ALTER TABLE foo ALTER id ADD GENERATED BY DEFAULT AS IDENTITY');
```
When running this migration, I get:
* A new sequence `foo_id_seq1`, additional to `foo_id_seq` (with "Current Value": 31) I already have.
* The ID column changes from Constraint Type "None" to "Identity", with "Start": 1

Then, obviously, when persisting a new entity, I'm getting:

> An exception occurred while executing a query: SQLSTATE[23505]: Unique violation: 7 ERROR: duplicate key value violates unique constraint "foo_pkey"
DETAIL: Key (id)=(1) already exists.

## Unclear Recommendations

Before upgrading, I got this deprecation:

> User Deprecated: Relying on non-optimal defaults for ID generation is deprecated, and IDENTITY results in SERIAL, which is not recommended. Instead, configure identifier generation strategies explicitly through configuration. We currently recommend "SEQUENCE" for "Doctrine\DBAL\Platforms\PostgreSqlPlatform", so you should use $configuration->setIdentityGenerationPreferences([ "Doctrine\DBAL\Platforms\PostgreSqlPlatform" => ClassMetadata::GENERATOR_TYPE_SEQUENCE, ]); (ClassMetadataFactory.php:755 called by ClassMetadataFactory.php:629, https://github.com/doctrine/orm/issues/8893, package doctrine/orm)

Which (basically) says: Switch to `SEQUENCE`!

On the other hand, https://github.com/doctrine/orm/blob/3.0.x/UPGRADE.md#bc-break-auto-keyword-for-identity-generation-defaults-to-identity-for-postgresql-when-using-doctrinedbal-4 says:

> When using the AUTO strategy to let Doctrine determine the identity generation mechanism for an entity [...] PostgreSQL now uses IDENTITY instead of SEQUENCE.

Which (basically) says: Doctrine recommends `IDENTITY`.

## Solution?

So the questions are:
* Which one is the recommended strategy now? SEQUENCE or IDENTITY?
* Since (more or less) everybody will run into the above migration problem sooner or later: Is there a chance that the migration generator can be improved to take care of this? If not, the upgrade instructions need to be improved to tell people how to proceed (I could take care of this).

### PR DESCRIPTION
**Pull request by ThomasLandauer**
Closes https://github.com/doctrine/orm/issues/11248

### CODE REVIEW

**Review by jwage — DISMISSED** [review]

File: UPGRADE.md [hunk 1]
```diff
@@ -13,9 +13,9 @@ so `$targetEntity` is a first argument now. This change affects only non-named a
 
 When using the `AUTO` strategy to let Doctrine determine the identity generation mechanism for
 an entity, and when using `doctrine/dbal` 4, PostgreSQL now uses `IDENTITY`
-instead of `SEQUENCE`. When upgrading from ORM 2.x and preference is on keeping
-the `SEQUENCE` based identity generation, then configure the ORM this way:
-
+instead of `SEQUENCE`.
+* If you want to upgrade your existing tables to `IDENTITY`, see [Migration to identity columns on PostgreSQL](https://www.doctrine-project.org/projects/doctrine-dbal/en/4.0/how-to/postgresql-identity-migration.html)
+* If you want to keep `SEQUENCE` or `SERIAL`, then configure the ORM this way:
```
Comment (greg0ire) [inline UPGRADE.md]: :thinking: I understand what you mean by "keeping `SEQUENCE`", but what do you mean by "keeping `SERIAL`"?

Comment (ThomasLandauer) [reply UPGRADE.md]: Maybe it's not clear to everybody that types SEQUENCE and SERIAL are the same (at least I didn't know that). I now added "type" - I guess that's what you mean, right?

Comment (greg0ire) [reply UPGRADE.md]: I think there are 3 concepts:

1. Identifier generation strategies (`AUTO`, `IDENTITY`, `SEQUENCE`, etc.)
2. What SQL object they map to (a database sequence, an auto-increment column, an identity column).
3. The DDL used to create these objects (`SERIAL`, which is no longer used with ORM 3 and DBAL 4, `GENERATED ALWAYS AS IDENTITY`, etc.)

I think people are very confused about this, so let's try hard not to increase the confusion, by not using these 3 concepts interchangeably.

```rst
* If you want to upgrade your existing tables to identity columns, you will need
to follow [migration to identity columns on PostgreSQL](https://www.doctrine-project.org/projects/doctrine-dbal/en/4.0/how-to/postgresql-identity-migration.html)
* If you want to keep using SQL sequences, you need to configure the ORM this way:

**Review by greg0ire — APPROVED** [review]

File: UPGRADE.md (hunk 1 above)
Comment (ThomasLandauer) [reply UPGRADE.md]: OK, take a look now.

**Review by jwage — APPROVED** [review]

File: UPGRADE.md (hunk 1 above)
Comment (greg0ire) [reply UPGRADE.md]: I did, and approved 👍
Just keeping that conversation open in case other maintainers disagree with my suggestion.

**Review by SenseException — APPROVED** [review]

**Pull request merged**